### DISCUSSION TITLE
EMQX reporting wrong RAM in ECS Fargate

### DISCUSSION THREAD

**Discussion by Frtrillo** [emqx/emqx#13978] [Q&A] [answered]
Im running EMQX on ECS Fargate, but RAM isn't reporting correctly and I fear that EMQX mismanages it? Or could be an ECS Fargate issue...
For example, I got up:

2 nodes with 3 gb ram: shows 3.75 G
2 nodes with 1 gb RAM: shows 1.81 G
![image](https://github.com/user-attachments/assets/76a0825d-28c0-44e2-8b4c-4b2a580220df)

**Comment by ieQu1** [comment]
EMQX relies on the OS to report free memory and CPU load. Certain container runtimes are known to report invalid data. We have special workarounds for docker, but Fargate may be different.

**Comment by id** [comment, marked as answer]
Here are 2 articles diving deep into the issue.
TL;DR, there is no standard way of getting namespaced RAM and CPU usage from inside a container.
The second article is from 2014, but it's still mostly accurate when it comes to getting accurate metrics in this scenario.

* https://ops.tips/blog/why-top-inside-container-wrong-memory/
* https://fabiokung.com/2014/03/13/memory-inside-linux-containers/

**Reply by Frtrillo** [reply]
Oh i see, thanks. Well I dont mind as long as its only a stat in the dashboard. I guess emqx and underlying kernel manages the RAM properly aside of how much RAM is "reading"

**Reply by id** [reply]
correct
